app/src/common/components/Button.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+
+import {
+  Button as MuiButton,
+  ButtonProps,
+  withStyles,
+} from "@material-ui/core";
+import { Typography } from "@material-ui/core";
+
+export const EditedButton = withStyles((theme) => ({
+  root: {
+    textTransform: "none",
+  },
+  startIcon: {
+    height: 12,
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    fill: theme.palette.common.white,
+  },
+  containedPrimary: {
+    border: `1px solid ${theme.palette.primary.dark}`,
+  },
+  containedSecondary: {
+    border: `1px solid ${theme.palette.secondary.dark}`,
+    "&:hover": {
+      border: `1px solid ${theme.palette.secondary.dark}`,
+    },
+    "& .MuiButton-startIcon": {
+      fill: theme.palette.secondary.contrastText,
+    },
+  },
+  disabled: {
+    border: "none",
+  },
+  colorInherit: {
+    "& .MuiButton-startIcon": {
+      fill: theme.palette.text.secondary,
+    },
+    color: theme.palette.text.secondary,
+    "&:hover": {
+      backgroundColor: "inherit",
+      color: theme.palette.text.primary,
+      "& .MuiButton-startIcon": {
+        fill: theme.palette.text.primary,
+      },
+    },
+  },
+}))(MuiButton);
+
+const Button = ({ ...buttonProps }: ButtonProps): JSX.Element => {
+  return (
+    <EditedButton disableElevation {...buttonProps}>
+      <Typography>{buttonProps.children}</Typography>
+    </EditedButton>
+  );
+};
```

Comment:
Be careful, you're injecting `buttonProps.children` twice in the component. You can "remove" `children` from the `buttonProps` object this way :

```suggestion
const Button = ({ children, ...buttonProps }: ButtonProps): JSX.Element => {
  return (
    <EditedButton disableElevation {...buttonProps}>
      <Typography>{children}</Typography>
    </EditedButton>
  );
};
```